internal/internal_workflow_client.go

```diff
@@ -1121,14 +1121,21 @@ func (workflowRun *workflowRunImpl) Get(ctx context.Context, valuePtr interface{
 		err = newTerminatedError()
 	case eventpb.EventType_WorkflowExecutionTimedOut:
 		attributes := closeEvent.GetWorkflowExecutionTimedOutEventAttributes()
-		err = NewTimeoutError(attributes.GetTimeoutType())
+		err = NewTimeoutError(attributes.GetTimeoutType(), nil)
 	case eventpb.EventType_WorkflowExecutionContinuedAsNew:
 		attributes := closeEvent.GetWorkflowExecutionContinuedAsNewEventAttributes()
 		workflowRun.currentRunID = attributes.GetNewExecutionRunId()
 		return workflowRun.Get(ctx, valuePtr)
 	default:
-		err = fmt.Errorf("unexpected event type %s when handling workflow execution result", closeEvent.GetEventType())
+		return fmt.Errorf("unexpected event type %s when handling workflow execution result", closeEvent.GetEventType())
 	}
+
+	err = NewWorkflowExecutionError(
+		workflowRun.workflowID,
+		workflowRun.currentRunID,
+		getWorkflowFunctionName(workflowRun.registry, workflowRun.workflowFn),
+		err)
```

Comment:
In case of workflow execution error, `WorkflowExecutionError` is returned with actual error inside.